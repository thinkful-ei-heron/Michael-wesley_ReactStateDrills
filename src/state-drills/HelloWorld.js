import React from 'react';


class HelloWorld extends React.Component{
  state = {
    who: 'world'
  }
  handleClick = arg => {
    this.setState({
      who: arg
    })
  }

  render(){
    return (
      <div>
        <p>Hello, {this.state.who}</p>
        <button type="button" onClick={() => this.handleClick('world')}>World</button>
        <button type="button" onClick={() => this.handleClick('friend')}>Friend</button>
        <button type="button" onClick={() => this.handleClick('React')}>React</button>
      </div>
    )
  }
}

export default HelloWorld


