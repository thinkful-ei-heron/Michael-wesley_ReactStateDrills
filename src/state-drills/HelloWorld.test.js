import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import HelloWorld from './HelloWorld';



describe('helloworld test', () => { 
  it('renders without crashing', () => {
    const result = document.createElement('div');
    ReactDOM.render(<HelloWorld/>, result);
    ReactDOM.unmountComponentAtNode(result);
  })
  it('renders as expected', ()=>{
    expect(renderer.create(<HelloWorld />)
    .toJSON())
    .toMatchSnapshot()
  })

})