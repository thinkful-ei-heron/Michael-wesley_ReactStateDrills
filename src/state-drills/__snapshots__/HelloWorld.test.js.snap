// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`helloworld test renders as expected 1`] = `
<div>
  <p>
    Hello, 
    world
  </p>
  <button
    onClick={[Function]}
    type="button"
  >
    World
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    Friend
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    React
  </button>
</div>
`;
